BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -43,53 +57,508 @@ public struct AttributedLabel: Element, Hashable {
+
+        override func touchesCancelled(_ touches: Set<UITouch>, with event: UIEvent?) {
+            trackingLinks = nil
+            applyLinkColors()
+        }
     }
+
 }
 
+extension AttributedLabel {
+    struct Link: Equatable, Hashable {
+        var text: String
+        var range: NSRange
+    }
+
+    private final class LinkAccessibilityElement: UIAccessibilityElement {
+        private let link: AttributedLabel.Link
+        private var container: LabelView? { accessibilityContainer as? LabelView }
+
+        init(
+            container: LabelView,
+            label: String,
+            link: AttributedLabel.Link
+        ) {
+            self.link = link
+            super.init(accessibilityContainer: container)
+            accessibilityLabel = label
+        }
+
+        override var accessibilityFrameInContainerSpace: CGRect {
+            set { fatalError() }
+            get {
+                guard let container = container,
+                      let textStorage = container.makeTextStorage(),
+                      let layoutManager = textStorage.layoutManagers.first,
+                      let textContainer = layoutManager.textContainers.first
+                else {
+                    return .zero
+                }
+
+                let glyphRange = layoutManager.glyphRange(forCharacterRange: link.range, actualCharacterRange: nil)
+                return layoutManager.boundingRect(forGlyphRange: glyphRange, in: textContainer)
+            }
+        }
+
+        override func accessibilityActivate() -> Bool {
+            container?.linkHandler?.open(link: link.text)
+            return true
+        }
+    }
+}
+
+// MARK: Environment
+
+/// Conform to this protocol to handle links tapped in an `AttributedLabel`.
+///
+/// Use the `LinkHandlerEnvironmentKey` or `Environment.linkHandler` property to override
+/// the link handler in the environment.
+///
+public protocol LinkHandler {
+    func open(link: String)
+}
+
+struct DefaultLinkHandler: LinkHandler {
+    func open(link: String) {
+        if let url = URL(string: link) {
+            UIApplication.shared.open(url, options: [:], completionHandler: nil)
+        }
+    }
+}
+
+public struct LinkHandlerEnvironmentKey: EnvironmentKey {
+    public static var defaultValue: LinkHandler {
+        DefaultLinkHandler()
+    }
+}
+
+extension Environment {
+    /// The link handler to use to open links tapped in an `AttributedLabel`.
+    public var linkHandler: LinkHandler {
+        get { self[LinkHandlerEnvironmentKey.self] }
+        set { self[LinkHandlerEnvironmentKey.self] = newValue }
+    }
+}
+
+struct ClosureLinkHandler: LinkHandler {
+    var openLink: (String) -> Void
+
+    func open(link: String) {
+        openLink(link)
+    }
+}
+
+extension AttributedLabel {
+    /// Handle links opened in the receiver using the provided closure.
+    ///
+    public func openLink(_ closure: @escaping (String) -> Void) -> Element {
+        AdaptedEnvironment(
```

Comment:
Minor nit: Would be a bit more readable here to use

`self.adaptedEnvironment(\.linkHandler, ClosureLinkHandler(openLink: closure))`?

Reply:
- [ ] Improve wrapper